src/taipy/core/job/_job_converter.py

```diff
@@ -42,7 +43,11 @@ def _model_to_entity(cls, model: _JobModel) -> Job:
         task_repository = task_manager._repository
 
         job = Job(
-            id=model.id, task=task_repository._load(model.task_id), submit_id=model.submit_id, version=model.version
+            id=model.id,
+            task=task_repository._load(model.task_id),
+            submit_id=model.submit_id,
+            submit_entity_id=model.submit_entity_id,
```

Comment:
Maybe we can hold the `submit_entity` in the `Job` and only store the `submit_entity_id` in the model. Exactly like we did for the job.task.

Reply:
I believe because the `id` attribute isn't available in the Submittable class

Reply:
sure I don't see why not :D